### PR TITLE
[Communication] - [Package] - Exclude ObjC files 

### PR DESCRIPTION
**Pull request by JoshuaLai**
Exclude the obj-c in package, and include swift build command into the CI

### CODE REVIEW

**Comment by JoshuaLai** [comment]
https://github.com/Azure/azure-sdk-for-ios/issues/570
An idea to fix the swift build.

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml [hunk 1]
```diff
@@ -37,7 +37,11 @@ jobs:
       - publish: $(Build.ArtifactStagingDirectory)/Build/Products/Release-iphonesimulator
         displayName: 'Publish release artifacts [iphonesimulator]'
         artifact: 'bundles-release-iphonesimulator'
-
+      
+      - script: |
+          swift build
+        displayName: "Use swift build for SPM"
+
```
Comment (JoshuaLai) [inline eng/pipelines/templates/jobs/archetype-sdk-client.yml:44]: Not sure if this script is enough or needs more details. Or needs to be in its own job, currently I just put it under the existing build job

Comment (tjprescott) [reply eng/pipelines/templates/jobs/archetype-sdk-client.yml:44]: It should be sufficient. If you exclude one of your exclusions does it fail the CI?

Comment (JoshuaLai) [reply eng/pipelines/templates/jobs/archetype-sdk-client.yml:44]: Created commit b469331 where I removed some files and you can see the CI failed for that case so looks like it is working as expected.

File: Package.swift [hunk 2]
```diff
@@ -71,6 +71,12 @@ let package = Package(
             name: "AzureCommunicationTests",
             dependencies: ["AzureCommunication"],
             path: "sdk/communication/AzureCommunication",
+            exclude: [
+                "Tests/AzureCommunicationTests-Bridging-Header.h",
+                "Tests/ObjCCommunciationUserCredentialTests.m",
+                "Tests/ObjCCommunicationUserCredentialAsyncTests.m",
+                "Tests/ObjCTokenParserTests.m"
+            ],
```
Comment (tjprescott) [inline Package.swift:79]: Is there a way to wildcard exclude *.m and *.h?  If not, this is okay for now.

**Review by tjprescott — COMMENTED** [review]
LGTM though I would prefer the exclude statement to be more generic. Would also prefer @bsiegel or @mitchdenny  approve.

File: Package.swift (hunk 2 above)
Comment (JoshuaLai) [reply Package.swift:79]: @tjprescott Doing a bit more research on this. But according to this post from Jun 22 2020 https://forums.swift.org/t/support-for-wildcard-paths-in-package-swift/37814 it looks like it was pitched but not able to yet.

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml [hunk 3]
```diff
@@ -37,7 +37,11 @@ jobs:
       - publish: $(Build.ArtifactStagingDirectory)/Build/Products/Release-iphonesimulator
         displayName: 'Publish release artifacts [iphonesimulator]'
         artifact: 'bundles-release-iphonesimulator'
-
+      
+      - script: |
```
Comment (bsiegel) [inline eng/pipelines/templates/jobs/archetype-sdk-client.yml:41]: I would prefer this be in its own job, that way we can run Xcode and SPM builds in parallel.

**Review by tjprescott — CHANGES_REQUESTED** [review]
Can't merge this until I understand why `/azp run` fails.

**Comment by tjprescott** [comment]
@bsiegel ADO is not running the pipeline on this PR. Any ideas why?

**Comment by bsiegel** [comment]
> @bsiegel ADO is not running the pipeline on this PR. Any ideas why?

I'm not positive but I believe the job YAML for "Swift Build" is invalid. Let me try pushing a change to fix that.

**Comment by JoshuaLai** [comment]
> > @bsiegel ADO is not running the pipeline on this PR. Any ideas why?
> 
> I'm not positive but I believe the job YAML for "Swift Build" is invalid. Let me try pushing a change to fix that.

Could it be a formatting issue? Spaces are off?

**Comment by bsiegel** [comment]
> Could it be a formatting issue? Spaces are off?

A couple things:

1. Spacing was off for 'displayName'
2. The job name is not allowed to contain spaces

**Review by tjprescott — APPROVED** [review]

**Pull request merged**